transport/http2_client.go

```diff
@@ -763,11 +767,19 @@ func (t *http2Client) Write(s *Stream, data []byte, opts *Options) error {
 			t.writableChan <- 0
 			continue
 		}
-		if r.Len() == 0 && t.framer.adjustNumWriters(0) == 1 {
+		if r.Len() == 0 && t.framer.adjustNumWriters(0) == 1 && isLastSlice {
```

Comment:
How about moving this after the new if `r.Len() == 0` that you added so you don't need to check `isLastSlice`?